python/ray/data/_internal/block_batching/iter_batches.py

```diff
@@ -239,7 +243,9 @@ def before_epoch_start(self):
         self._yielded_first_batch = False
 
     def after_epoch_end(self):
-        StatsManager.clear_iteration_metrics(self._dataset_tag)
+        _StatsManager.update_iteration_metrics(
+            self._stats, self._dataset_tag, force_update=True
+        )
```

Comment:
My understanding is that, in the previous implementation, we don't update any metrics in `after_epoch_end`.

Don't think it's an issue, but what's the motivation for performing a force update here?